src/jit/inline.cpp

```diff
@@ -1120,83 +1168,159 @@ bool InlineStrategy::s_DumpDataHeader = false;
+                    limit);
+            fprintf(stderr, "Method,Version,HotSize,ColdSize,JitTime,SizeEstimate,TimeEstimate");
+            m_LastSuccessfulPolicy->DumpSchema(stderr);
             fprintf(stderr, "\n");
         }
+
+        s_DumpDataHeader = true;
+    }
+
+    // We'd really like the method identifier to be unique and
+    // durable across crossgen invocations. Not clear how to
+    // accomplish this, so we'll use the token for now.
+    //
+    // Post processing will have to filter out all data from
+    // methods where the root entry appears multiple times.
+    mdMethodDef currentMethodToken =
+        info.compCompHnd->getMethodDefFromMethod(info.compMethodHnd);
+
+    // Convert time spent jitting into microseconds
+    unsigned microsecondsSpentJitting = 0;
+    unsigned __int64 compCycles = m_Compiler->getInlineCycleCount();
+    if (compCycles > 0)
+    {
+        double countsPerSec = CycleTimer::CyclesPerSecond();
+        double counts = (double) compCycles;
+        microsecondsSpentJitting = (unsigned) ((counts / countsPerSec) * 1000 * 1000);
+    }
+
+    fprintf(stderr,
+            "%08X,%u,%u,%u,%u,%d,%d",
+            currentMethodToken,
+            m_InlineCount,
+            info.compTotalHotCodeSize,
+            info.compTotalColdCodeSize,
+            microsecondsSpentJitting,
+            m_CurrentSizeEstimate / 10,
+            m_CurrentTimeEstimate);
+    m_LastSuccessfulPolicy->DumpData(stderr);
+    fprintf(stderr, "\n");
+}
+
+//------------------------------------------------------------------------
+// DumpXML: dump xml-formatted version of the inline tree.
+//
+// Arguments
+//    file - file for data output
+//    indent - indent level of this element
+
+void InlineStrategy::DumpXml(FILE* file, unsigned indent)
+{
+    if (JitConfig.JitInlineDumpXml() == 0)
+    {
+        return;
     }
+
+    // Cache references to compiler substructures.
+    const Compiler::Info& info = m_Compiler->info;
+    const Compiler::Options& opts = m_Compiler->opts;
+
+    const bool isPrejitRoot = (opts.eeFlags & CORJIT_FLG_PREJIT) != 0;
+    const bool isForceInline = (info.compFlags & CORINFO_FLG_FORCEINLINE) != 0;
+
+    // We'd really like the method identifier to be unique and
+    // durable across crossgen invocations. Not clear how to
+    // accomplish this, so we'll use the token for now.
+    //
+    // Post processing will have to filter out all data from
+    // methods where the root entry appears multiple times.
+    mdMethodDef currentMethodToken =
+        info.compCompHnd->getMethodDefFromMethod(info.compMethodHnd);
+
+#ifdef DEBUG
+    unsigned hash = info.compMethodHash();
+#else
+    unsigned hash = 0;
+#endif
+
+    // Convert time spent jitting into microseconds
+    unsigned microsecondsSpentJitting = 0;
+    unsigned __int64 compCycles = m_Compiler->getInlineCycleCount();
+    if (compCycles > 0)
+    {
+        double countsPerSec = CycleTimer::CyclesPerSecond();
```

Comment:
Various checks over on the compiler side ensure `compCycles` is zero if the cycles per second can't be determined. See for instance `Compiler::RecordStateAtEndOfCompilation`